File: core/models/Donation.py
import core.models.User as User
from datetime import date

class Donation:

    def __init__(self, id, donor: User, donationCampaignId: str, amount: int):
        self.__id = id
        self.__amount = amount
        self.__dateDonated = date.today()
        self.__donationCampaign = donationCampaignId
        self.__donor = donor

    
    def getId(self):
        return self.__id

    def getAmount(self):
        return self.__amount
    
    def getDateDonated(self):
        return self.__dateDonated.isoformat()
    
    def getDonationCampaign(self):
        return self.__donationCampaign
    
    def getDonor(self):
        return self.__donor
    
    def getData(self):
        return {
            "amount": self.getAmount(),
            "dateDonated": self.getDateDonated(),
            "donationCampaignId": self.getDonationCampaign(),
            "donor": self.getDonor()
        }
        

File: src/app/models/Donation.ts


export class Donation{

    constructor(
        private id: string,
        private amount: number,
        private dateDonated: string,
        private donationCampaignId: string,
        private donor: string
    ){}

    public getId(){
        return this.id
    }

    public getAmount(){
        return this.amount
    }

    public getDateDonated(){
        return this.dateDonated
    }

    public getDonationCampaignId(){
        return this.donationCampaignId
    }

    public getDonor(){
        return this.donor
    }
}